drools-game-engine-horserace-kjar/src/main/resources/rules/game/gamerules.drl

```diff
@@ -0,0 +1,76 @@
+package org.drools.game.horserace;
+import org.drools.game.core.api.*;
+import org.drools.game.model.api.*;
+import org.drools.game.horserace.model.*;
+
+global GameMessageService messageService;
+global GameCallbackService callback;
+
+rule "Greeter"
+    when
+        $player : Player()
+    then
+        insert(  messageService.newGameMessage(  "Welcome, " + $player.getName() + ". Venture forth to the stables and mount a mighty steed!" ) );
+end
+
+rule "Give the player a team"
+    when
+        $player : Player()
+        not( Team( name == $player.name) ) 
+    then
+        Team team = new Team($player.getName());
+        team.addPlayer($player.getName());
+        insert(team);
+end
+
+rule "first checkpoint"
+        when
+            $player : Player()
+            $team : Team(name == $player.name && points == 0)
+            $room : Room(id == "CheckPointOne" && playersInRoom contains $player.getName())
```

Comment:
We should have generic rules, as soon as you reference the ID of the room in the rule, that rule stops being generic. So my question is, can we make this Rule generic for all the checkpoints?